### PR TITLE
added Oops-solidity summary

### PR DESCRIPTION
**Pull request by omm-prakash**

### CODE REVIEW

File: src/summaries/chapter5.yaml [hunk 1]
```diff
@@ -0,0 +1,136 @@
+- title: Understanding Encapsulation in Solidity
+  shortTitle: Encapsulation
+  key: encapsulation
+  details: |
+    It refers to the bundling of data, along with the methods that operate on that data, into a single unit.
+    It refers to the mechanism of manipulating the scope of variables, i.e., restricting the variable's access outside the scope.
+  
+    Syntax
+      ```
```
Comment (RobinNagpal) [inline src/summaries/chapter5.yaml]: its single word
`     ```solidity    `

File: src/summaries/chapter5.yaml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  
+    - Private
+      * Private objects can only be accessed internally from the current contract instances.
+      * They cannot be accessed by derived methods also.
+  
+    - External
+      * These objects can be accessed externally but not internally, i.e., current contract instances cannot access it.
+      * They cannot be inherited.
+    
+    State Variable visibility
```
Comment (RobinNagpal) [inline src/summaries/chapter5.yaml:34]: Also mention one or two use cases i.e. 

* Use public for variables when you want the to be visible outside

and so on

File: src/summaries/chapter5.yaml [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  details: |
+    Inheritance is the process of defining multiple contracts related to each other through parent-child relationships.
+  
+    Solidity supports inheritance between smart contracts.
+      * Base contract: The contract from which other contracts inherit features.
+      * Derived contract: The contract which inherits the features.
+      * Simply, they are referred to as parent-child contracts.
+    
+    Syntax
+      ```
```
Comment (RobinNagpal) [inline src/summaries/chapter5.yaml]: its single word
`     ```solidity    `

File: src/summaries/chapter5.yaml [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+ 
+- title: Polymorphism in Solidity
+  shortTitle: Polymorphism
+  key: polymorphism
+  details: |
+    It is an ability to process data in more than one form.
+  
+    Solidity supports two types of polymorphism.
+      - Function polymorphism
+        * It is also known as method overloading.
```
Comment (RobinNagpal) [inline src/summaries/chapter5.yaml:96]: Give example also

File: src/summaries/chapter5.yaml [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+      solidity
+      constructor() <Access Modifier> {         
+      }
+      ```
+  
+    Note
+      * It is very useful in a smart contract, a parameter value can be defined at the run time and can also restrict the method call.
+      * Constructor overloading is not supported in Solidity; it only allows one constructor at a time.
+ 
+- title: Abstraction in Solidity
```
Comment (RobinNagpal) [inline src/summaries/chapter5.yaml:194]: We need to also cover libraries and interfaces

https://www.tutorialspoint.com/solidity/solidity_interfaces.htm
https://www.tutorialspoint.com/solidity/solidity_libraries.htm

File: src/summaries/chapter5.yaml [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+    Note : Solidity follows the path of Python and uses C3 Linearization, also known as Method Resolution Order (MRO), to force a specific order in graphs of base contracts. The order of execution is determined by order of the declaration at the contract level. It has nothing to do with the order you call/pass arguments up from the constructor's signature.
+ 
+- title: Polymorphism in Solidity
+  shortTitle: Polymorphism
+  key: polymorphism
+  details: |
+    It is an ability to process data in more than one form.
+  
+    Solidity supports two types of polymorphism.
+      - Function polymorphism
```
Comment (RobinNagpal) [inline src/summaries/chapter5.yaml:95]: We need to explain both overloading and overriding functions

File: src/summaries/chapter5.yaml (hunk 2 above)
Comment (omm-prakash) [reply src/summaries/chapter5.yaml:34]: Use cases are mentioned in the detail section

File: src/summaries/chapter5.yaml (hunk 5 above)
Comment (omm-prakash) [reply src/summaries/chapter5.yaml:194]: Interface is explained in the abstraction details section

**Pull request merged**